member/src/main/proto/com/improving/app/member/domain/improving-app-member-domain.proto

```diff
@@ -0,0 +1,198 @@
+  MEMBER_STATUS_INITIAL = 0;
+  MEMBER_STATUS_ACTIVE = 1;
+  MEMBER_STATUS_INACTIVE = 2;
+  MEMBER_STATUS_SUSPENDED = 3;
+  MEMBER_STATUS_TERMINATED = 4;
+}
+
+message MemberInfo {
+  option (scalapb.message).extends = "com.improving.app.member.utils.serialize.CborSerializable";
+  string handle = 1;
+  string avatar_url = 2;
+  string first_name = 3;
+  string last_name = 4;
+  optional string mobile_number = 5;
+  optional string email = 6;
+  NotificationPreference notification_preference = 7;
+  bool notification_opt_in = 8;
+  repeated com.improving.app.organization.api.OrganizationId organizations = 9;
+  string related_members = 10;
+  repeated MemberType member_types = 11;
+}
+
+message MemberMap {
+  option (scalapb.message).extends = "com.improving.app.member.utils.serialize.CborSerializable";
+  MemberId member_id = 1;
+  MemberInfo member_info = 2;
+}
+
+message MemberMetaInfo {
+  option (scalapb.message).extends = "com.improving.app.member.utils.serialize.CborSerializable";
+  sint64 created_on = 1;
+  MemberId created_by = 2;
+  sint64 last_modified_on = 3;
+  MemberId last_modified_by = 4;
+  MemberStatus member_state = 5;
+}
+
+//Requests
+message RegisterMember {
+  MemberInfo member_info = 1;
+  MemberId registering_member = 2;
+}
+
+message ActivateMember {
+  MemberId member_id = 1;
+  MemberId acting_member = 2;
+}
+
+message InactivateMember {
+  MemberId member_id = 1;
+  MemberId acting_member = 2;
+}
+
+message SuspendMember {
+  MemberId member_id = 1;
+  MemberId acting_member = 2;
+}
+
+message TerminateMember {
+  MemberId member_id = 1;
+  MemberId acting_member =2;
+}
+
+message UpdateMemberInfo {
+  MemberId member_id = 1;
+  MemberInfo member_info = 2;
+  MemberId acting_member = 3;
+}
+
+message GetMemberInfo {
+  MemberId member_id = 1;
+}
+
+message MemberRequest{
+  option (scalapb.message).sealed_oneof_extends = "com.improving.app.member.utils.serialize.CborSerializable";
+
+  oneof sealed_value {
+    RegisterMember register_member_value = 1;
+    ActivateMember activate_member_value = 2;
+    InactivateMember inactivate_member_value = 3;
+    SuspendMember suspend_member_value = 4;
+    TerminateMember terminate_member_value = 5;
+    UpdateMemberInfo update_member_info_value = 6;
+    GetMemberInfo get_member_info_value = 7;
+  }
+}
+
+//Member Events
+message MemberRegistered {
+  MemberId member_id = 1;
+  MemberInfo member_info =2;
+  MemberMetaInfo member_meta_info = 3;
+}
+
+message MemberActivated {
+  MemberId member_id = 1;
+  MemberMetaInfo member_meta =2;
+}
+
+message MemberInactivated {
+  MemberId member_id = 1;
+  MemberMetaInfo member_meta =2;
+}
+
+message MemberSuspended {
+  MemberId member_id = 1;
+  MemberMetaInfo member_meta = 2;
+}
+
+message MemberTerminated {
+  MemberId member_id = 1;
+  MemberMetaInfo member_meta = 2;
+}
+
+message MemberInfoUpdated {
+  MemberId member_id = 1;
+  MemberInfo member_info = 2;
+  MemberMetaInfo member_meta_info = 3;
+}
+
```

Comment:
I doubt that the consumers of those events need all that member_info and all the member_meta_info. Remember, they have the member_id and they can call back to the member service to get details. Some commonly needed information (name, contact info) should be transmitted with the event. They are also missing timestamps of when the event was generated, "Timestamp eventTime", which is often needed.  Think about the usage of these events.  Note that the storage events for persistence might be different from these events that travel to, typically, the command issuer.

Reply:
Added Timestamp to the Events